src/mds/Server.cc

```diff
@@ -1433,7 +1433,7 @@ void Server::reply_client_request(MDRequestRef& mdr, MClientReply *reply)
     req->get_connection()->send_message(reply);
   }
 
-  if (req->is_queued_for_replay() &&
+  if (req->is_queued_for_replay() && reply &&
```

Comment:
Is there a way to convert this into an `assert(reply)` that makes clang happy?

Reply:
sure, as long as we are sure that `reply` is always valid here.

Reply:
hmm, after a second thought, i am not qualified to do the conversion as i don't understand mds enough to figure out where "here" is. will drop this commit.